desk/app/reel.hoon

```diff
@@ -221,7 +301,126 @@
+      ?.  ?=(%self -.response)  `this
+      =*  profile  con.response
+      =>
+        |%
+        ::  +hand: choose right and test equality
+        ++  hand
+          |*  [a=(unit) b=(unit)]
+          ^-  [_b ?]
+          ?~  a  [b ?=(~ b)]
+          ?~  b  [~ ?=(~ a)]
+          [b =(u.a u.b)]
+        --
+      ::  check profile for relevant changes
+      ::
+      =/  [nickname=(unit @t) axe=?]
+        %+  hand
+          (~(get cy:t our-profile) %nickname %text)
+        (~(get cy:t profile) %nickname %text)
+      =/  [avatar=(unit @t) bax=?]
+        %+  hand
+          (~(get cy:t our-profile) %avatar %look)
+        (~(get cy:t profile) %avatar %look)
+      =/  [color=(unit @ux) cax=?]
+        %+  hand
+          (~(get cy:t our-profile) %color %tint)
+        (~(get cy:t profile) %color %tint)
+      =.  our-profile  profile
+      ::  nothing relevant has changed, skip the update
+      ?:  &(axe bax cax)  `this
+      =|  update=metadata:reel
+      =.  tag.update  'group-0'
+      =.  fields.update
+        %-  ~(gas by *(map cord cord))
+        :~  'inviterNickname'^(fall nickname '')
+            'inviterAvatarImage'^(fall avatar '')
+            'inviterColor'^?^(color (rsh [3 2] u.color) '')
+        ==
+      ::  update our lure links with new nickname and avatar image
+      ::
+      =.  our-metadata
+        %-  ~(run by our-metadata)
+        |=  meta=metadata:reel
+        meta(fields (~(uni by fields.meta) fields.update))
+      ::  request the bait provider to update our invite links
+      ::
+      =/  caz=(list card)
+        %+  turn  ~(tap by our-metadata)
+        |=  [=token:reel *]
+        [%pass /update/contact %agent [civ %bait] %poke bait-update+!>([token update])]
+      [caz this]
+    ==
+  ::
+      [%groups ~]
+    ?+    -.sign  (on-agent:def wire sign)
+        %kick
+      =^  caz=(list card)  subs
+        (~(subscribe s [subs bowl]) /groups [our.bowl %groups] groups-path &)
+      [caz this]
+    ::
+        %watch-ack
+      ?~  p.sign  `this
+      %-  (fail:log %watch-ack 'failed to subscribe to groups' u.p.sign)
+      `this
+    ::
+        %fact
+      =+  !<(=r-groups:v7:groups-ver q.cage.sign)
+      ?.  ?=(%meta -.r-group.r-groups)  `this
+      =*  flag  flag.r-groups
+      =*  meta  meta.r-group.r-groups
+      =+  id=(rap 3 (scot %p p.flag) '/' q.flag ~)
+      ?~  token=(~(get by stable-id) id)  `this
+      =|  update=metadata:reel
+      =.  tag.update  'groups-0'
+      =.  fields.update
+        %-  ~(gas by *(map cord cord))
+        :~  'invitedGroupTitle'^title.meta
+            'invitedGroupDescription'^description.meta
+            'invitedGroupIconImageUrl'^image.meta
+        ==
+      ::  update our group invite link
+      ::
+      =.  our-metadata
+        ?~  our-meta=(~(get by our-metadata) u.token)
+          our-metadata
+        %+  ~(put by our-metadata)  u.token
+        u.our-meta(fields (~(uni by fields.u.our-meta) fields.update))
```

Comment:
I don't remember/understand what this gets used for on the reel side. Do we track this exclusively for the "changed link providers" case (`%set-service`)?